MdeModulePkg/Core/Dxe/Misc/MemoryAttributesTable.c

```diff
@@ -447,18 +447,23 @@ EnforceMemoryMapAttribute (
   MemoryMapEntry = MemoryMap;
   MemoryMapEnd   = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)MemoryMap + MemoryMapSize);
   while ((UINTN)MemoryMapEntry < (UINTN)MemoryMapEnd) {
-    switch (MemoryMapEntry->Type) {
-      case EfiRuntimeServicesCode:
-        // do nothing
-        break;
-      case EfiRuntimeServicesData:
-      case EfiMemoryMappedIO:
-      case EfiMemoryMappedIOPortSpace:

```

Comment:
Why remove setting for EfiMemoryMappedIO and EfiMemoryMappedIOPortSpace?

Reply:
@lgao4, that came from Ard in the original review of this on the mailing list, see https://edk2.groups.io/g/devel/topic/patch_v1_mdemodulepkg/105570114?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,105570114. There is backing for it in the UEFI spec:

```text
The Memory Attributes Table is currently used to describe memory protections that may be applied to the EFI Runtime code and data by an operating system or hypervisor.
Consumers of this table must currently ignore entries containing any values for Type except for EfiRuntimeServicesData
and EfiRuntimeServicesCode to ensure compatibility with future uses of this table.
```

From UEFI spec 2.10 section 4.6.3 EFI_MEMORY_ATTRIBUTES_TABLE.

Reply:
I suggest to add the separate commit to remove the setting for EfiMemoryMappedIO and EfiMemoryMappedIOPortSpace because they are not in MAT.

Reply:
@lgao4, I have split this into two commits, can you please re-review?